src/apps/relay/mainrelay.c

```diff
@@ -2779,1082 +2968,1086 @@ static void print_features(unsigned long mfn) {
+                  "Be aware that you could not mix the username/password and the shared secret based auth methods. \n"
+                  "Shared secret overrides username/password based auth method. Check your configuration!\n");
+  }
 
-    if (!(curve_name[0])) {
-#if !SSL_SESSION_ECDH_AUTO_SUPPORTED
-      curve_name = DEFAULT_EC_CURVE_NAME;
-#endif
-      set_auto_curve = 1;
+  if (turn_params.allow_loopback_peers) {
+    TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING,
+                  "CONFIG: allow_loopback_peers opens a possible security vulnerability. Do not use in production!!\n");
+    if (cli_password[0] == 0 && use_cli) {
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR,
+                    "CONFIG: allow_loopback_peers and empty cli password cannot be used together.\n");
+      return -1;
     }
+  }
 
-    if (curve_name[0]) {
-      {
-        nid = OBJ_sn2nid(curve_name);
-        if (nid == 0) {
-          TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "unknown curve name: %s\n", curve_name);
-          curve_name = DEFAULT_EC_CURVE_NAME;
-          nid = OBJ_sn2nid(curve_name);
-          set_auto_curve = 1;
-        }
-      }
+  if (!use_lt_credentials && !anon_credentials) {
+    if (turn_params.default_users_db.ram_db.users_number) {
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING,
+                    "CONFIG: you specified long-term user accounts, (-u option) \n	but you did "
+                    "not specify the long-term credentials option\n	(-a or --lt-cred-mech option).\n 	I am "
+                    "turning --lt-cred-mech ON for you, but double-check your configuration.\n");
+      turn_params.ct = TURN_CREDENTIALS_LONG_TERM;
+      use_lt_credentials = 1;
+    } else {
+      turn_params.ct = TURN_CREDENTIALS_NONE;
+      use_lt_credentials = 0;
+    }
+  }
 
-      {
-        EC_KEY *ecdh = EC_KEY_new_by_curve_name(nid);
-        if (!ecdh) {
-          TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "%s: ERROR: allocate EC suite\n", __FUNCTION__);
-          set_auto_curve = 1;
-        } else {
-          SSL_CTX_set_tmp_ecdh(ctx, ecdh);
-          EC_KEY_free(ecdh);
-        }
-      }
+  if (use_lt_credentials) {
+    if (!get_realm(NULL)->options.name[0]) {
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING,
+                    "CONFIG: you did specify the long-term credentials usage\n but you did not specify "
+                    "the default realm option (-r option).\n		Check your configuration.\n");
     }
+  }
 
-    if (set_auto_curve) {
-#if SSL_SESSION_ECDH_AUTO_SUPPORTED
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
-      SSL_CTX_set_ecdh_auto(ctx, 1);
-#endif
-#endif
-      set_auto_curve = 0;
+  if (anon_credentials) {
+    if (turn_params.default_users_db.ram_db.users_number) {
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING,
+                    "CONFIG: you specified user accounts, (-u option)	but you also specified the "
+                    "anonymous user access option (-z or --no-auth option).	User accounts will be ignored.\n");
```

Comment:
Instead of ignoring the user accounts flag, this combination should be considered a hard error, and the process should shutdown.